include/llbuild/Core/BuildDB.h

```diff
@@ -138,11 +138,20 @@ class BuildDB {
 /// \param clientSchemaVersion An uninterpreted version number for use by the
 /// client to allow batch changes to the stored build results; if the stored
 /// schema does not match the provided version the database will be cleared upon
-/// opening.
+/// opening; to avoid this behavior, use `createNondestructiveSQLiteBuildDB`.
 std::unique_ptr<BuildDB> createSQLiteBuildDB(StringRef path,
                                              uint32_t clientSchemaVersion,
                                              std::string* error_out);
 
+/// Create a BuildDB instance backed by a SQLite3 database.
+/// If the given client schema version doesn't match the one of the database,
+/// an error will be thrown at use instead of automatically re-creating the
+/// database. This behavior might be useful if the database's content is important
+/// instead of the caching functionality.
+std::unique_ptr<BuildDB> createNondestructiveSQLiteBuildDB(StringRef path,
+                                                           uint32_t clientSchemaVersion,
+                                                           std::string* error_out);
```

Comment:
This feels like it should be a parameter to `createSQLiteBuildDB`, not an extra method?

Reply:
Yep, that's whatI had before. But I wanted to give it a default value which is only possible if the error_out parameter has a default value as well. Would you rather have it as a parameter without a default value then?